packages/core/src/external/mpi/normalize-patient.ts

```diff
@@ -0,0 +1,164 @@
+      ...contact,
+      email: contact.email ? normalizeEmail(contact.email) : contact.email,
+      phone: contact.phone ? normalizePhoneNumber(contact.phone) : contact.phone,
+    })),
+    address: (patient.address ?? []).map(addr => {
+      const newAddress: Address = {
+        addressLine1: normalizeAddress(addr.addressLine1),
+        city: normalizeString(addr.city),
+        zip: addr.zip.slice(0, 5),
+        state: addr.state,
+        country: addr.country || "USA",
+      };
+      if (addr.addressLine2) {
+        newAddress.addressLine2 = normalizeAddress(addr.addressLine2);
+      }
+      return newAddress;
+    }),
+  };
+  return handleDefaultValues(normalizedPatient);
+}
+
+/**
+ * The normalizeString function takes a string as input, removes leading and trailing whitespace,
+ * converts all characters to lowercase, and removes any apostrophes or hyphens.
+ * @param {string} str - The `str` parameter is a string that represents the input string that needs to
+ * be normalized.
+ * @returns a normalized version of the input string.
+ */
+function normalizeString(str: string): string {
+  return str.trim().toLowerCase().replace(/['-]/g, "");
+}
+
+/**
+ * Normalizes an email address by removing leading and trailing spaces, converting all characters to lowercase,
+ * and handling common domain typos.
+ *
+ * @param email - The email address to be normalized.
+ * @returns The normalized email address.
+ */
+function normalizeEmail(email: string): string {
+  return email.trim().toLowerCase();
+}
+
+/**
+ * Normalizes a phone number by removing all non-numeric characters and, if applicable, removing the country code.
+ * @param phoneNumber - The phone number to be normalized.
+ * @returns The normalized phone number as a string.
+ */
+function normalizePhoneNumber(phoneNumber: string): string {
+  const normalizedNumber = phoneNumber.replace(/\D/g, "");
+
+  if (normalizedNumber.startsWith("1") && normalizedNumber.length === 11) {
+    // applies to US and Canada numbers only
+    return normalizedNumber.substring(1);
+  }
+
+  return normalizedNumber;
+}
+
+/**
+ * The function `normalizeAddress` takes a string representing an address and replaces common street
+ * suffixes with their abbreviated forms.
+ * @param {string} address - The `address` parameter is a string that represents a street address.
+ * @returns The function `normalizeAddress` returns a string.
+ */
+function normalizeAddress(address: string): string {
+  const suffixes: Record<string, string> = {
+    street: "st",
+    avenue: "ave",
+    boulevard: "blvd",
+    drive: "dr",
+    road: "rd",
+    terrace: "ter",
+    place: "pl",
+    lane: "ln",
+    court: "ct",
+    circle: "cir",
+    highway: "hwy",
+    parkway: "pkwy",
+  };
+
+  address = address.trim().toLowerCase().replace(/['-.]/g, "");
+  const words: string[] = address.split(" ");
+
+  const transformedWords = words.map(word => {
+    const suffix = suffixes[word];
```

Comment:
just googled `Circle Ave` and theres a bunch of those throughout the states. Need to think of a way to avoid this scenario cause this could potentially ruin patient matching. Same with `Court Drive`. 

The rest are probably ok..

Reply:
that is a very good point

Reply:
yah I had a comment on [that](https://github.com/metriport/metriport/pull/1160#discussion_r1394280387) about trying to integrate the work @RamilGaripov did on that here. 
But then that brings us back to the question of when normalization happens, since doing AWS location services over and over again on the same data is wasteful from a cost peerspective

Reply:
we could just not normalise addresses for now since it seems to be the value that has the most complexity.